src/Identity/IdentityServer/BaseRequestValidator.cs

```diff
@@ -610,7 +610,7 @@ private async Task UpdateFailedAuthDetailsAsync(User user, bool twoFactorInvalid
 
         if (ValidateFailedAuthEmailConditions(unknownDevice, user))
         {
-            if (twoFactorInvalid)
+            if (twoFactorInvalid) //Todo need test path for here
```

Comment:
It was not b/c the `UserManager<User>` was giving me trouble when I was trying to mock/substitute values.

@jlf0dev and I opted to ditch the unit tests for integration tests in this instance. The `BaseRequestValidator` two factor flows are tested in the [companion PR ](https://github.com/bitwarden/server/pull/4643). Integration testing is a better approach since there is desire to refactor out the two factor parts which means the test would have been "lost work" in a sense. 

Anyway, I'll remove the comment.